### PR TITLE
Zlib4libpng

### PR DESCRIPTION
**Pull request by lecorguille**
My apologies

libpng need CPPFLAGS and LDFLAGS from zlib during the compilation.

You may use a sexier method like autoconf to do that but I didn't manage to do without using shell_command.

### CODE REVIEW

File: packages/package_libpng_1_6_7/tool_dependencies.xml [hunk 1]
```diff
@@ -12,7 +12,10 @@
             <package name="zlib" version="1.2.8" />
           </repository>
         </action>
-        <action type="autoconf"/>
+        <action type="shell_command">
+                    ./configure CPPFLAGS="-I$ZLIB_ROOT_PATH/include" LDFLAGS="-L$ZLIB_ROOT_PATH/lib" --prefix="$INSTALL_DIR"
+        </action>
+        <action type="make_install" />
```
Comment (bgruening) [inline packages/package_libpng_1_6_7/tool_dependencies.xml]: Can you try:
<action type="autoconf">CPPFLAGS="-I$ZLIB_ROOT_PATH/include" LDFLAGS="-L$ZLIB_ROOT_PATH/lib" </action>

Comment (lecorguille) [inline packages/package_libpng_1_6_7/tool_dependencies.xml]: @bgruening :+1:

**Comment by bgruening** [comment]
@lecorguille does it work for you?

**Comment by lecorguille** [comment]
I tested from the testtoolshed and it's seem good: https://testtoolshed.g2.bx.psu.edu/view/lecorguille/package_libpng_1_6_7/d7c48620e72a

One more test with a really poor VM and I confirm that

**Comment by lecorguille** [comment]
I confirm that it's better to include FLAGS when your server already have a version of zlib installed.

**Comment by bgruening** [comment]
Anyone else what to review this?

**Comment by yhoogstrate** [comment]
After installing from @lecorguille's repo I get indeed the following lines in the INSTALLATION.log:
```
gcc -E -DHAVE_CONFIG_H -I. \
    -I<galaxy-dir>/tool-dependencies/zlib/1.2.8/iuc/package_zlib_1_2_8/03abe628fbe8/include -DPNGLIB_LIBNAME='PNG16_0' -DPNGLIB_VERSION='1.6.7' -DSYMBOL_PREFIX='' -DPNG_NO_USE_READ_MACROS -DPNG_BUILDING_SYMBOL_TABLE  pnglibconf.c > pnglibconf.tf1
...
ibtool: compile:  gcc -DHAVE_CONFIG_H -I. -I<galaxy-dir>/tool-dependencies/zlib/1.2.8/iuc/package_zlib_1_2_8/03abe628fbe8/include -g -O2 -MT pngmem.lo -MD -MP -MF .deps/pngmem.Tpo -c pdepbase=`echo pngpread.lo | sed 's|[^/]*$|.deps//bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.   -I<galaxy-dir>/tool-dependencies/zlib/1.2.8/iuc/package_zlib_1_2_8/03abe628fbe8/include  -g -O2 -MT pngpread.lo -MD -MP -MF $depbase.Tpo -c -o pngpread.lo pngpread.c &&\
```
which were missing in installations from the iuc and devteam repo's. Compilation goes is fine so I think we can merge, but probably devteam's repo must be updated as well?

**Comment by lecorguille** [comment]
I will propose the same PR.
But just a stupid question, why choose to maintain 2 versions of some tools/packages?

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**